From a797b79483940ed4adcaa5fe2c40dd0487c7c2c7 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 9 Aug 2022 11:39:08 -0700
Subject: [PATCH 4/5] Use correct type to store return from flb_kv_item_create

Fix
error: incompatible pointer to integer conversion assigning to 'int' from 'struct flb_kv *'

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
Upstream-Status: Pending

 plugins/out_stackdriver/stackdriver_conf.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plugins/out_stackdriver/stackdriver_conf.c b/plugins/out_stackdriver/stackdriver_conf.c
index a9a8eb0..e4f969e 100644
--- a/plugins/out_stackdriver/stackdriver_conf.c
+++ b/plugins/out_stackdriver/stackdriver_conf.c
@@ -176,12 +176,12 @@ static int read_credentials_file(const char *cred_file, struct flb_stackdriver *
 
 static int parse_configuration_labels(struct flb_stackdriver *ctx)
 {
-    int ret;
     char *p;
     flb_sds_t key;
     flb_sds_t val;
     struct mk_list *head;
     struct flb_slist_entry *entry;
+    struct flb_kv *ret;
     msgpack_object_kv *kv = NULL;
 
     if (ctx->labels) {
@@ -216,7 +216,7 @@ static int parse_configuration_labels(struct flb_stackdriver *ctx)
             flb_sds_destroy(key);
             flb_sds_destroy(val);
 
-            if (ret == -1) {
+            if (!ret) {
                 return -1;
             }
         }
-- 
2.37.1

